### PR TITLE
22750 - JWT token refresh leads to HTTP 403: Access denied

### PR DESCRIPTION
**Pull request by fscz**
This PR adds some slack to the function that decides if a JWT token is expired. 
See https://github.com/istio/istio/issues/22750 
for details

### CODE REVIEW

File: security/pkg/server/ca/authenticate/authenticator.go [hunk 1]
```diff
@@ -101,7 +102,7 @@ func NewIDTokenAuthenticator(aud string) (*IDTokenAuthenticator, error) {
 		return nil, err
 	}
 
-	verifier := provider.Verifier(&oidc.Config{ClientID: aud})
+	verifier := provider.Verifier(&oidc.Config{ClientID: aud, Now: func() time.Time { return time.Now().Add(-30 * time.Second) }})
```
Comment (yangminzhu) [inline security/pkg/server/ca/authenticate/authenticator.go:105]: How is this related to https://github.com/istio/istio/issues/22750?

Comment (fscz) [reply security/pkg/server/ca/authenticate/authenticator.go:105]: It's not. The actual code is in https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/jwt_authn/authenticato.cc, line 240:
const Status status = ::google::jwt_verify::verifyJwt(*jwt_, *jwks_data_->getJwksObj());.

Closing this.
